Mathlib/LinearAlgebra/Matrix/NonsingularInverse.lean

```diff
@@ -361,6 +362,92 @@ lemma mul_right_injective_of_inv (A : Matrix m n α) (B : Matrix n m α) (h : A
 
 end InjectiveMul
 
+section vecMul
+
+variable [Fintype m] [DecidableEq m] {K : Type*} [Field K]
+
+theorem mulVec_surjective_iff_exists_rightInverse {A : Matrix m n α} :
+    Function.Surjective A.mulVec ↔ ∃ B : Matrix n m α, A * B = 1 := by
+  refine ⟨fun h ↦ ?_, fun ⟨B, hBA⟩ y ↦ ⟨B.mulVec y, by simp [hBA]⟩⟩
+  have hch : ∀ i : m, ∃ x, A.mulVec x = Pi.single i 1 := fun _ ↦ h _
+  choose cols hcols using hch
+  use (Matrix.of cols).transpose
+  apply_fun Matrix.transpose using transpose_injective
+  apply funext fun j ↦ ?_
+  convert hcols j
+  ext i
+  rw [Pi.single_apply]
+  rfl
+
+theorem vecMul_surjective_iff_exists_leftInverse {A : Matrix m n α} :
+    Function.Surjective A.vecMul ↔ ∃ B : Matrix n m α, B * A = 1 := by
+  simp_rw [← mulVec_transpose, mulVec_surjective_iff_exists_rightInverse]
+  refine ⟨fun ⟨B, hB⟩ ↦ ⟨Bᵀ, ?_⟩, fun ⟨B, hB⟩ ↦ ⟨Bᵀ, by rw [←transpose_mul, hB, transpose_one]⟩⟩
+  apply_fun Matrix.transpose using transpose_injective
+  rw [transpose_mul, transpose_transpose, hB, transpose_one]
+
+theorem vecMul_surjective_iff_isUnit {A : Matrix m m α} :
+    Function.Surjective A.vecMul ↔ IsUnit A := by
```

Comment:
I think it might be a little cleaner to prove 
```suggestion
    (∃ B : Matrix n m α, B * A = 1) ↔ IsUnit A := by
```
which could go right next to `isUnit_of_left_inverse`.